### ISSUE TITLE
Interactive window is lost when reconnecting to a remote project

### ISSUE THREAD

**Issue by jraviotta** [microsoft/vscode-jupyter#8760]
I have a project on a remote server. I use my local workspace to open the remote via ssh. Our University enforces a timeout on the connection to the server. When I reconnect after refreshing the ssh tunnel, my interactive window is lost. (I am using .py versions of jupyter notebooks with the # %% notation for cells.)

I want the interactive window to be available when I reconnect so that I can continue working on my analysis without having to rerun the entire workbook.

**Comment by roblourens** [comment]
When the window reloads, a new remote extension host is created, and a new kernel instance will be created. Did we talk about this issue at some point @DonJayamanne? I thought there might be another issue for this scenario but I can't find it.

**Comment by DonJayamanne** [comment]
This is a duplicate of https://github.com/microsoft/vscode-jupyter/issues/3998

This is by design, when VS Code is loaded (the old kernels are disposed) & the kernels are re-created. I.e. life time of kernels is currently tied to a VS Code session.

**The only work around for now is:**
* On SSH machine, start jupyter server manually
* Connect to jupyter using the Remote connections in the command palette `Jupyter: Specifcy local or remote Jupyter server connections`
* Start a kernel

Once this is done, we treat the kernel as remote & never dispose the kernel.
FYI - Please try this in VS Code insiders & pre-release version of the Jupyter extension.

**Comment by jraviotta** [comment]
Thanks for the work around. I look forward to a more permanent resolution.
